### DISCUSSION TITLE
Tailwind/babel-plugin-preval with SWC

### DISCUSSION THREAD

**Discussion by Parnswir** [vercel/next.js#31841] [Help]
We want to upgrade a medium-size Nextjs project to use SWC with Next12, which works fine so far, except for one small issue: We have set up TailwindCSS and need to reference parts of it's configuration within components. Following [Tailwind's docs](https://tailwindcss.com/docs/configuration#referencing-in-java-script), we have setup [babel-plugin-preval](https://github.com/kentcdodds/babel-plugin-preval), to reduce the bundle size by generating the merged Tailwind config during the build step and thus excluding Tailwind's dev dependencies from the final bundle.

With SWC, the above approach does not work and as a consequence our First Load JS grows by more than 20%. After a pretty demotivating few hours of trying to make this last piece of the puzzle work, I'm now hoping to get some pointers from the community on how to approach this. It's hard for me to believe nobody else seems to be using babel-plugin-preval in https://github.com/vercel/next.js/discussions/30174.

**Comment by draxx318** [comment]
I'm having the same issue. Did you manage to fix this?

**Reply by Parnswir** [reply]
Not really. If your config overrides in the tailwind config do not depend on the theme (i.e. they are plain objects instead of `(theme) => ({})` functions), you can extract them into a file and import it from both the tailwind config and your components. If you do need the theme, I don't see how to do it right now.

**Reply by drewlustro** [reply]
As far as I know, Vercel is slowly implementing the most popular babel plugins in rust.

Given that preval is a fairly advanced use-case, I doubt it's high on their list.

**Reply by Tosinkoa** [reply]
Please do you manage to get this to work?

**Reply by fonzarely** [reply]
Did someone test that one ? 
https://www.npmjs.com/package/next-plugin-preval
It seems to work fine for me.

**Reply by fonzarely** [reply]
My preval file if need be
```js
import preval from 'next-plugin-preval'

import resolveConfig from 'tailwindcss/resolveConfig'
import tailwindConfig from 'tailwind.config'

async function getData() {
  const fullConfig = resolveConfig(tailwindConfig)
  //   console.log('PREVAL fullConfig', fullConfig.theme.screens)
  return fullConfig.theme.screens
}

export default preval(getData())

```
